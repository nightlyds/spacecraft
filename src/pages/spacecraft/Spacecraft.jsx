import React, { useEffect } from 'react'
import PropTypes from 'prop-types';
import MenuButton from './components/menu/MenuButton.jsx'
import counterpart from 'counterpart';
import en from './lang/en';
import uk from './lang/uk';

counterpart.registerTranslations('en', en);
counterpart.registerTranslations('uk', uk);
counterpart.setLocale('en');

const Spacecraft = ({ loadUp, data, loading, error }) => {
    // useEffect(() => {
    //     loadUp();
    // }, [])
    return (
        <div className="spacecraft">
            <MenuButton />
        </div>
    )
}

Spacecraft.propTypes = {
    loadUp: PropTypes.func,
    data: PropTypes.array,
    loading: PropTypes.bool,
    error: PropTypes.bool
}

Spacecraft.defaultProps = {
    loadUp: () => {},
    data: [],
    loading: false,
    error: false
}

export default Spacecraft;
