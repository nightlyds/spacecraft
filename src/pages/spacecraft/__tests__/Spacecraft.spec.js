import { shallow } from "enzyme";
import React from "react";
import Spacecraft from "../Spacecraft.jsx";

const setUp = (props) => shallow(<Spacecraft {...props} />);

describe("Spacecraft tests", () => {
  describe("One wrapper element test", () => {
    let component;

    beforeEach(() => {
      component = setUp();
    });

    it("Wrapper element test", () => {
      const wrapper = component.find(".spacecraft");
      expect(wrapper.length).toBe(1);
    });
  });

  describe("Snapshot tests", () => {
    let component;

    beforeEach(() => {
      component = setUp();
    });

    it("Default snapshot", () => {
      expect(component).toMatchSnapshot();
    });

    it("With loading true and data", () => {
      component = setUp({ loading: true, data: {0: "0", 1: "1" }, error: false });
      expect(component).toMatchSnapshot();
    });

    it("With loading true and error true", () => {
      component = setUp({ loading: true, error: true });
      expect(component).toMatchSnapshot();
    });
  });

  describe("Props test", () => {
    it("Data props test", () => {
      let result = Spacecraft.defaultProps.data;
      expect(result).toEqual({});
    });

    it("Loading props test", () => {
        let result = Spacecraft.defaultProps.loading;
        expect(result).toEqual(false);
      });

    it("Error props test", () => {
      let result = Spacecraft.defaultProps.error;
      expect(result).toEqual(false);
    });

    it("LoadUp props test", () => {
        let result = Spacecraft.defaultProps.loadUp();
        expect(result).toEqual(undefined);
      });
  });
});
