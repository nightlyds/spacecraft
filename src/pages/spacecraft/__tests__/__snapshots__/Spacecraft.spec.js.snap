// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacecraft tests Snapshot tests Default snapshot 1`] = `
<div
  className="spacecraft"
>
  <MenuButton
    icon={false}
    menuOpen={[Function]}
  />
  <LanguageChange />
  <div
    className="spacecraft-content"
  >
    <Loading />
  </div>
  <Footer />
</div>
`;

exports[`Spacecraft tests Snapshot tests With loading true and data 1`] = `
<div
  className="spacecraft"
>
  <MenuButton
    icon={false}
    menuOpen={[Function]}
  />
  <LanguageChange />
  <div
    className="spacecraft-content"
  >
    <GraphicCost
      data={
        Object {
          "0": "0",
          "1": "1",
        }
      }
    />
    <Description />
  </div>
  <Footer />
</div>
`;

exports[`Spacecraft tests Snapshot tests With loading true and error true 1`] = `
<div
  className="spacecraft"
>
  <MenuButton
    icon={false}
    menuOpen={[Function]}
  />
  <LanguageChange />
  <div
    className="spacecraft-content"
  >
    <Error />
  </div>
  <Footer />
</div>
`;
